ProcessMaker/processmaker
# Enhancement: Centralized Ellipsis Action Menus Configuration

## Issue & Reproduction Steps
This PR introduces a central configuration area via the `ellipsisMenuActions.js` file, serving as a mixin to manage ellipsis menu actions for Processes, Scripts, and Screens within packages. This file provides a single point for adding, updating, or removing actions related to these assets.

Additionally, each asset type (Process, Script, Screen) now includes an associated mixin file (`processNavigation.js`, `screenNavigation.js`, `scriptNavigation.js`) to control methods that facilitate when specific actions are triggered. These mixin files can also be utilized within packages to extend ellipsis actions as needed.

All of these files have been implemented within the 'shared' components, making them accessible for use within packages by importing from SharedComponents.

## Key Changes:

- Refactoring of action button filtering based on conditionals and permissions.
- Restructuring of ellipsis action and event handling.

## How to Test

1. For each asset type (Processes, Screens, Scripts), test the ellipsis menu on the respective designer page.
2. Create a project and assign the above-mentioned assets.
3. Ensure that actions for these assets are enabled or disabled based on permissions and conditions, and that they are displayed alongside project-specific actions such as 'Copy to Project' and 'Assign to Project.'
4. Confirm that all actions function as expected.

## Related Tickets & Packages
- [FOUR-11019](https://processmaker.atlassian.net/browse/FOUR-11019)
- Projects PR

ci:next

## Code Review Checklist
